logzio/datasource_metrics_account.go

```
@@ -0,0 +1,117 @@
+	"strconv"
+	"strings"
+	"time"
+)
+
+func dataSourceMetricsAccount() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMetricsAccountReadWrapper,
+		Schema: map[string]*schema.Schema{
+			metricsAccountId: {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			metricsAccountName: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountEmail: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountToken: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountPlanUts: {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			metricsAccountAuthorizedAccounts: {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+				Optional: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Second),
+		},
+	}
+}
+
+func dataSourceMetricsAccountReadWrapper(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var err error
+	readErr := retry.Do(
+		func() error {
+			// we use this instead of return dataSourceMetricsAccountRead(d, m) directly
+			// to avoid 'no "id" found in attributes' error
+			if err = dataSourceMetricsAccountRead(d, m); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		retry.RetryIf(
+			func(err error) bool {
+				if err != nil {
+					if strings.Contains(err.Error(), "failed with missing metrics account") ||
+						strings.Contains(err.Error(), "failed with status code 500") {
+						return true
+					}
+				}
+				return false
+			}),
+		retry.DelayType(retry.BackOffDelay),
+		retry.Attempts(15),
+	)
+
+	if readErr != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func dataSourceMetricsAccountRead(d *schema.ResourceData, m interface{}) error {
+	var client *metrics_accounts.MetricsAccountClient
+	client, _ = metrics_accounts.New(m.(Config).apiToken, m.(Config).baseUrl)
```

Why this error is ignored?